### ISSUE TITLE
kube-router enters CrashLoopBackOff state when API Server is down

### ISSUE THREAD

**Issue by sbrohi-att** [cloudnativelabs/kube-router#739]
**kube-router version: 0.3.1**
**kubernetes version: 1.11.9**

We have noticed that kube-router enters a "CrashLoopBackOff" state when the kube-apiservers are down. 

The routers eventually stabilize once the API Server is up, but we need to allow kube-router to continue functioning and preserving routes even in the absence of losing the api server. 

Below are the errors we see flooding the logs: 
```
E0522 21:49:49.361459       1 reflector.go:205] github.com/cloudnativelabs/kube-router/vendor/k8s.io/client-go/informers/factory.go:73: Failed to list *v1.NetworkPolicy: Get https://mymasters.com:6443/apis/networking.k8s.io/v1/networkpolicies?resourceVersion=0: dial tcp <master-node-ip>:6443: connect: connection refused
E0522 21:49:50.357801       1 reflector.go:205] github.com/cloudnativelabs/kube-router/vendor/k8s.io/client-go/informers/factory.go:73: Failed to list *v1.Service: Get https://mymasters.com:6443/api/v1/services?resourceVersion=0: dial tcp <master-node-ip>:6443: connect: connection refused
```

We also added the following flag to kube-router, but this did not help: 
```
--bgp-graceful-restart
```

In one of the crashed pods, we also found the following: 
```
kube-router Network Routing Controller heartbeat missed
```

Is there any configuration tuning / workarounds we can apply to ensure kube-router runs and preserves routes when the API Server is down? We need to be able to run kube-router while the API Server is down for several hours.

**Comment by danbeaulieu** [comment]
Also seeing this on a 1.13.5 cluster. While kuberouter is crashing pods do not have any network connectivity.

**Comment by rdegez** [comment]
This is affecting us as well.

IMHO an unreachable apiserver event happening with an already started and running kube-router instance should stick in a retry-loop, keeping every installed ip routes (& maybe iptables rules) in place.

This would give a chance to the workload running on worker nodes to stay up while the k8s control-plane is fixed back to a clean state.

**Comment by andrewsykim** [comment]
Agreed, kube-router should run independent of API server state. Will investigate soon

**Comment by aauren** [comment]
Closed via #862
